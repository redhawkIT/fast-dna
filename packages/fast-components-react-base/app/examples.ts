import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };
