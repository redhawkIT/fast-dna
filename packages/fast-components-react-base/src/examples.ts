import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: Array<IExampleData<T>>;
}
