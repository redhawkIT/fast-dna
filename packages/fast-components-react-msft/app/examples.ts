import ButtonExample from "../src/button/examples.data";
export {ButtonExample};

import DividerExample from "../src/divider/examples.data";
export {DividerExample};

import ToggleExamples from "../src/toggle/examples.data";
export {ToggleExamples};
