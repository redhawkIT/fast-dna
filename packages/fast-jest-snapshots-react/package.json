{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "version": "1.6.0",
  "author": {
    "name": "Microsoft",
    "email": "fwq@microsoft.com"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "private": true,
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "prepare": "tsc -p ./tsconfig.json",
    "test": "npm run tslint",
    "start": "webpack-dev-server",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'"
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^1.6.0",
    "@types/jest": "^22.2.2",
    "jest": "^22.4.3",
    "react": "^16.3.0",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^2.8.1"
  },
  "dependencies": {
    "@types/react": "^16.3.0",
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
