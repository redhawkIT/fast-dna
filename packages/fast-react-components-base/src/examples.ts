import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: IExampleData<T>[];
}