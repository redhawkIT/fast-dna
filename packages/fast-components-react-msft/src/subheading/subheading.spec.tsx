import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import MSFTSubheading, {
    ISubheadingHandledProps,
    ISubheadingManagedClasses,
    ISubheadingUnhandledProps,
    SubheadingLevel,
    SubheadingTag,
 } from "./subheading";
import { Subheading } from "./index";

 /**
  * Configure enzyme
  */
configure({adapter: new Adapter()});

describe("generate snapshots", (): void => {
    generateSnapshots(examples);
});

describe("subheading", (): void => {
    test("should have a displayName that matches the component name", () => {
        expect((MSFTSubheading as any).name).toBe(MSFTSubheading.displayName);
    });

    test("should return a valid object which includes all props that are not enumerated as handledProps", (): void => {
        const handledProps: ISubheadingHandledProps = {
            tag: SubheadingTag.h1,
            size: SubheadingLevel._1
        };

        const unhandledProps: ISubheadingUnhandledProps = {
            "aria-hidden": true
        };

        const props: ISubheadingHandledProps & ISubheadingUnhandledProps = {...handledProps, ...unhandledProps};

        const rendered: any = shallow(
            <Subheading {...props} />
        );

        const subheading: any = rendered.first().shallow();

        expect(subheading.prop("aria-hidden")).toEqual(true);
    });

    test("should render the correct `tag` when `tag` prop is passed in", (): void => {
        const rendered: any = shallow(
            <Subheading tag={SubheadingTag.h4} />
        );

        const subheading: any = rendered.first().shallow();

        expect(subheading.instance().props.tag).toEqual(SubheadingTag.h4);
    });

    test("should render default size if none is specified", (): void => {
        const rendered: any = shallow(
            <Subheading />
        );

        const subheading: any = rendered.first().shallow();

        expect(subheading.instance().props.size).toEqual(SubheadingLevel._1);
    });
});
