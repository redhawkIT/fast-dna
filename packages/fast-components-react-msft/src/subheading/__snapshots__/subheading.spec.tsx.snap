// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate snapshots Subheading: 0 1`] = `
<h3
  className="typography-0-1-1 typography__3-0-1-4 subheading-0-1-11 subheading__1-0-1-12"
>
  Subheading test string
</h3>
`;

exports[`generate snapshots Subheading: 1 1`] = `
<h1
  className="typography-0-1-1 typography__3-0-1-4 subheading-0-1-11 subheading__1-0-1-12"
  data-sketch-symbol="Subheading 1"
>
  Subheading test string
</h1>
`;

exports[`generate snapshots Subheading: 2 1`] = `
<h1
  className="typography-0-1-1 typography__3-0-1-4 subheading-0-1-11 subheading__1-0-1-12"
  data-sketch-symbol="Subheading 1"
>
  Subheading test string
</h1>
`;

exports[`generate snapshots Subheading: 3 1`] = `
<h2
  className="typography-0-1-1 typography__4-0-1-5 subheading-0-1-11 subheading__2-0-1-13"
  data-sketch-symbol="Subheading 2"
>
  Subheading test string
</h2>
`;

exports[`generate snapshots Subheading: 4 1`] = `
<h3
  className="typography-0-1-1 typography__5-0-1-6 subheading-0-1-11 subheading__3-0-1-14"
  data-sketch-symbol="Subheading 3"
>
  Subheading test string
</h3>
`;

exports[`generate snapshots Subheading: 5 1`] = `
<h4
  className="typography-0-1-1 typography__6-0-1-7 subheading-0-1-11 subheading__4-0-1-15"
  data-sketch-symbol="Subheading 4"
>
  Subheading test string
</h4>
`;

exports[`generate snapshots Subheading: 6 1`] = `
<h5
  className="typography-0-1-1 typography__7-0-1-8 subheading-0-1-11 subheading__5-0-1-16"
  data-sketch-symbol="Subheading 5"
>
  Subheading test string
</h5>
`;

exports[`generate snapshots Subheading: 7 1`] = `
<h6
  className="typography-0-1-1 typography__8-0-1-9 subheading-0-1-11 subheading__6-0-1-17"
  data-sketch-symbol="Subheading 6"
>
  Subheading test string
</h6>
`;
